frontend/ui/data/onetime_migration.lua

```diff
@@ -632,5 +632,20 @@ if last_migration_date < 20230901 then
     end
 end
 
+-- 20231212, change folder_shortcuts setting from array to hash table
+if last_migration_date < 20231212 then
+    logger.info("Performing one-time migration for 20231212")
+
+    local shortcuts = G_reader_settings:readSetting("folder_shortcuts")
+    if shortcuts and shortcuts[1] ~= nil then
```

Comment:
Not familiar with the data structure in question, but `ipairs` is perfectly safe on an empty table, so the `nil` guard is not necessarily needed (unless you really don't want to save an empty table in case things are really broken)?

Reply:
When testing, the purpose was to apply this onetime_migration step to already transformed shortcuts table (i.e. not to kill the hash table).
Not sure if such situation can occur in real life.

Reply:
Ooh, didn't think of that. Yeah, that works ;).